src/snowflake/connector/ocsp_snowflake.py

```diff
@@ -74,38 +74,50 @@ class OCSPResponseValidationResult(NamedTuple):
     validated: bool = False
 
 
-try:
-    OCSP_RESPONSE_VALIDATION_CACHE: SFDictFileCache[
-        tuple[bytes, bytes, bytes],
-        OCSPResponseValidationResult,
-    ] = SFDictFileCache(
-        entry_lifetime=constants.DAY_IN_SECONDS,
-        file_path={
-            "linux": os.path.join(
-                "~", ".cache", "snowflake", "ocsp_response_validation_cache"
-            ),
-            "darwin": os.path.join(
-                "~", "Library", "Caches", "Snowflake", "ocsp_response_validation_cache"
-            ),
-            "windows": os.path.join(
-                "~",
-                "AppData",
-                "Local",
-                "Snowflake",
-                "Caches",
-                "ocsp_response_validation_cache",
-            ),
-        },
-    )
-except OSError:
-    # In case we run into some read/write permission error fall back onto
-    #  in memory caching
-    OCSP_RESPONSE_VALIDATION_CACHE: SFDictCache[
-        tuple[bytes, bytes, bytes],
-        OCSPResponseValidationResult,
-    ] = SFDictCache(
-        entry_lifetime=constants.DAY_IN_SECONDS,
-    )
+# use _get_ocsp_response_validation_cache instead of the module variable for lazy evaluation
+OCSP_RESPONSE_VALIDATION_CACHE: SFDictFileCache[
+    tuple[bytes, bytes, bytes], OCSPResponseValidationResult
+] | SFDictCache[tuple[bytes, bytes, bytes], OCSPResponseValidationResult] | None = None
+
+
+def _get_ocsp_response_validation_cache():
+    global OCSP_RESPONSE_VALIDATION_CACHE
```

Comment:
This is an invisible side-effect of this function and usually a bad idea!

Reply:
oh, could you elaborate on this? I might be unfamiliar with the potential pitfalls here

Reply:
see comment above, change reverted